plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/maven2/metadata/MetadataRebuilder.java

```diff
@@ -238,23 +233,11 @@ public void end() {
      * Process exits from group level, executed in isolation.
      */
     private void rebuildMetadataExitGroup(final String currentGroupId) {
-      storageFacet.perform(db, new Operation<Void>()
```

Comment:
To clarify, for the plain `MetadataRebuilder` write/delete cases it doesn't need a retry here because `MavenFacetImpl` put and delete operations are already marked with `@Transactional` with retry enabled, so they will automatically retry using the current transaction context.

The update case in `MetadataRebuilder` is slightly different as we want to re-read before retrying the write, and with the `rebuildMetadataInner` case we want to repeat building the metadata for the base version instead of repeating the write with the same data.